rust/settings/BUILD.bazel

```diff
@@ -55,6 +60,16 @@ rust_lto_flag(
     visibility = ["//visibility:public"],
 )
 
+# The default value for `--codegen-units` which also affects resource
+# allocation for rustc actions. Note that any value 0 or less will prevent
+# this flag from being passed by Bazel.
+# https://doc.rust-lang.org/rustc/codegen-options/index.html#codegen-units
+int_flag(
+    name = "codegen_units",
+    build_setting_default = 16,
```

Comment:
I feel like we should default this to -1 or 1... Every codebase I've ever worked in has set codegen_units to 1 (at least for release builds, but generally by default) because it leads to better runtime optimisation, but -1 ("do what rustc fancies") seems like a reasonable default...

Reply:
Updated! Default is now `-1` which will prevent this flag from explicitly being passed and disable any `ctx.actions.run.resource_set` from being passed.